service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -55,11 +55,29 @@ export NXF_MODE=google
 
 sudo apt-get update
 
+#######################################
+# Install required JDK and set it as default (debian)
+#######################################
+function install_java() {
+  curl -Os https://download.oracle.com/java/17/latest/jdk-17_linux-x64_bin.deb
+  sudo apt-get install -y ./jdk-17_linux-x64_bin.deb
+  sudo update-alternatives --install /usr/bin/java java /usr/lib/jvm/jdk-17/bin/java 1
+  sudo update-alternatives --set java /usr/lib/jvm/jdk-17/bin/java
+}
+
 if [[ -n "$(which java)" ]];
 then
-  echo "java is installed"
+  readonly CUR_JAVA_VERSION=$(java -version 2>&1 | grep 'version' 2>&1 | awk -F\" '{ split($2,a,"."); print a[1]}')
```

Comment:
i think java -version 2>&1 | awk -F\" '{ split($2,a,"."); print a[1]}' is enough.
```
yuhuyoyo-macbookpro3% java -version 2>&1 | awk -F\" '{ split($2,a,"."); print a[1]}' 
17
```

Can we also add a comment saying that we strips out the decimal just because bash is hard to read..

Reply:
Done. added a comment - that only major version is considered